```
CC       libm4_a-clean-temp.o
  CC       libm4_a-clean-temp-simple.o
clean-temp.c:235:14: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
  234 |       error (0, errno,
      |       ~~~~~~~~~~~~~~~~
  235 |              _("cannot find a temporary directory, try setting $TMPDIR"));
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
clean-temp.c:69:18: note: expanded from macro '_'
   69 | #define _(msgid) dgettext ("gnulib", msgid)
      |                  ^
./gettext.h:94:39: note: expanded from macro 'dgettext'
   94 | #  define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
      |                                       ^
./error.h:496:39: note: expanded from macro 'error'
  496 |       __gl_error_call (error, status, __VA_ARGS__)
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
./error.h:432:44: note: expanded from macro '__gl_error_call'
  432 |      ? __gl_error_call1 (function, status, __VA_ARGS__)         \
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
./error.h:421:26: note: expanded from macro '__gl_error_call1'
  421 |     ((function) (status, __VA_ARGS__), \
      |                          ^~~~~~~~~~~
clean-temp.c:235:14: note: treat the string as an argument to avoid this
  235 |              _("cannot find a temporary directory, try setting $TMPDIR"));
      |              ^
      |              "%s",
clean-temp.c:69:18: note: expanded from macro '_'
   69 | #define _(msgid) dgettext ("gnulib", msgid)
      |                  ^
./gettext.h:94:39: note: expanded from macro 'dgettext'
   94 | #  define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
      |                                       ^
./error.h:496:39: note: expanded from macro 'error'
  496 |       __gl_error_call (error, status, __VA_ARGS__)
      |                                       ^
./error.h:432:44: note: expanded from macro '__gl_error_call'
  432 |      ? __gl_error_call1 (function, status, __VA_ARGS__)         \
      |                                            ^
./error.h:421:26: note: expanded from macro '__gl_error_call1'
  421 |     ((function) (status, __VA_ARGS__), \
      |                          ^
clean-temp.c:235:14: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
234 |       error (0, errno,
      |       ~~~~~~~~~~~~~~~~
  235 |              _("cannot find a temporary directory, try setting $TMPDIR"));
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
clean-temp.c:69:18: note: expanded from macro '_'
   69 | #define _(msgid) dgettext ("gnulib", msgid)
      |                  ^
./gettext.h:94:39: note: expanded from macro 'dgettext'
   94 | #  define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
      |                                       ^
./error.h:496:39: note: expanded from macro 'error'
  496 |       __gl_error_call (error, status, __VA_ARGS__)
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
./error.h:436:51: note: expanded from macro '__gl_error_call'
  436 |          __gl_error_call1 (function, __errstatus, __VA_ARGS__); \
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
./error.h:421:26: note: expanded from macro '__gl_error_call1'
  421 |     ((function) (status, __VA_ARGS__), \
      |                          ^~~~~~~~~~~
clean-temp.c:235:14: note: treat the string as an argument to avoid this
  235 |              _("cannot find a temporary directory, try setting $TMPDIR"));
      |              ^
      |              "%s",
clean-temp.c:69:18: note: expanded from macro '_'
   69 | #define _(msgid) dgettext ("gnulib", msgid)
      |                  ^
./gettext.h:94:39: note: expanded from macro 'dgettext'
   94 | #  define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
      |                                       ^
./error.h:496:39: note: expanded from macro 'error'
  496 |       __gl_error_call (error, status, __VA_ARGS__)
      |                                       ^
./error.h:436:51: note: expanded from macro '__gl_error_call'
  436 |          __gl_error_call1 (function, __errstatus, __VA_ARGS__); \
      |                                                   ^
./error.h:421:26: note: expanded from macro '__gl_error_call1'
  421 |     ((function) (status, __VA_ARGS__), \
      |                          ^
```

I don’t know why they’re excluding Clang here or really what on earth they’re doing in general but I can confirm that it does not work on Darwin.

--- a/pkgs/by-name/gn/gnum4/package.nix
+++ b/pkgs/by-name/gn/gnum4/package.nix
@@ -33,9 +33,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   enableParallelBuilding = true;
 
-  # Without the change the build fails on Darwin as https://hydra.nixos.org/build/300264726/nixlog/3/tail:
-  #   clean-temp.c:235:14: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
-  hardeningDisable = lib.optionals stdenv.hostPlatform.isDarwin [ "format" ];
+  # Issue exists whenever NLS is disabled, and there's an upstream fix
+  # for GCC, but there's no good way to check whether NLS or GCC is in
+  # use.  (Checking stdenv.cc.isGNU causes infinite recursion.)
+  hardeningDisable = [ "format" ];
 
   doCheck = false;
 

